File: activity/app/src/main/java/com/rikkei/training/activity/MediaManager.java
package com.rikkei.training.activity;

import android.content.Context;
import android.media.MediaPlayer;
import android.net.Uri;

public class MediaManager {
    private String link;
    private Context context;
    private MediaPlayer player;

    public MediaManager(String link, Context context) {
        this.link = link;
        this.context = context;
    }

    public void create() {
        Uri uri = Uri.parse(link);
        player = MediaPlayer.create(context, uri);
    }

    public void start() {
        if (player != null) {
            player.start();
        }
    }

    public void pause() {
        if (player != null) {
            player.pause();
        }
    }

    public void stop() {
        if (player != null) {
            player.stop();
        }
    }
}

File: Activitykotlin/app/src/main/java/com/rikkei/training/activitykotlin/MediaManager.kt
package com.rikkei.training.activitykotlin

import android.content.Context
import android.media.MediaPlayer
import android.net.Uri

class MediaManager {
    private var link = ""
    private var player: MediaPlayer? = null
    private var context: Context? = null

    constructor(context: Context?, link: String) {
        this.context = context
        this.link = link
    }

    fun create() {
        var uri: Uri = Uri.parse(link)
        player = MediaPlayer.create(context, uri)
    }

    fun start() {
        if (player != null) {
            player!!.start()
        }
    }

    fun pause() {
        if (player != null) {
            player!!.pause()
        }
    }

    fun stop() {
        if (player != null) {
            player!!.stop()
        }
    }
}